actions/actions.py

```diff
@@ -57,48 +149,45 @@ def backup() -> None:
     })
 
 
-def restore() -> None:
+def restore():
+    """
+    Implementation of easyrsa 'restore' action
+
+    Backup restoration process can be summarized as following:
+
+        * Selected backup is scanned and verified
+        * Contents of the backup are unpacked into <cahrm_dir>/EasyRSA/pki
+        * Data that are stored in the local database are updated
+        * All units that have relation with this easyrsa unit will be notified
+          about the certificate changes.
+    """
     pki_dir = os.path.join(easyrsa_directory, 'pki')
-    backup_name: str = function_get('name')
+    backup_name = function_get('name')
+
     if backup_name is None:
-        raise RuntimeError('Parameter \'name\' is required.')
+        raise RuntimeError("Parameter 'name' is required.")
 
     backup_path = os.path.join(PKI_BACKUP, backup_name)
 
-    shutil.rmtree(pki_dir, ignore_errors=True)
+    if not os.path.isfile(backup_path):
+        raise RuntimeError("Backup with name '{}' does not exist. Use action "
+                           "'list-backups' to list all available "
+                           "backups".format(backup_name))
+
     with tarfile.open(backup_path, 'r:gz') as tar:
-        try:
-            tar.extractall(easyrsa_directory)
-        except KeyError:
-            raise RuntimeError('Backup file "{}" does not contain expected '
-                               'files'.format(backup_name))
+        _verify_backup(tar)
+        _replace_pki(tar, pki_dir)
 
-    # Set these values on the leadership data.
     cert_dir = os.path.join(pki_dir, 'issued')
     key_dir = os.path.join(pki_dir, 'private')
-    ca_cert = os.path.join(pki_dir, 'ca.crt')
-    ca_key = os.path.join(key_dir, 'ca.key')
-    serial_file = os.path.join(pki_dir, 'serial')
-
-    with open(ca_cert, 'r') as stream:
-        print("Writing ca")
-        leader_set({
-            'certificate_authority': stream.read()})
-
-    with open(ca_key, 'r') as stream:
-        print("Writing key")
-        leader_set({
-            'certificate_authority_key': stream.read()})
 
-    with open(serial_file, 'r') as stream:
-        print("Writing serial")
-        leader_set({
-            'certificate_authority_serial': stream.read()})
+    _update_leadership_data(pki_dir, cert_dir, key_dir)
 
     ca_cert = leader_get('certificate_authority')
-    print(ca_cert)
     tls = endpoint_from_name('client')
     tls.set_ca(ca_cert)
+    tls.set_client_cert(leader_get('client_certificate'),
+                        leader_get('client_key'))
```

Comment:
Suggestion: use hookenv.log to report this relation data has been shared. Logging will help troubleshoot future issues by operations teams.